restapi/src/main/java/io/stargate/web/docsapi/service/ReactiveDocumentService.java

```diff
@@ -230,6 +230,66 @@ public ReactiveDocumentService(
         .map(any -> new DocumentResponseWrapper<>(documentId, null, null, context.toProfile()));
   }
 
+  /**
+   * Deletes a document with given ID in the given namespace and collection.
+   *
+   * @param db {@link DocumentDB} to write in
+   * @param namespace Namespace
+   * @param collection Collection name
+   * @param documentId The ID of the document to delete
+   * @param context Execution content @@return Flag representing if the operation was success.
+   */
+  public Single<Boolean> deleteDocument(
+      DocumentDB db,
+      String namespace,
+      String collection,
+      String documentId,
+      ExecutionContext context) {
+    List<String> subPath = Collections.emptyList();
+    return deleteDocument(db, namespace, collection, documentId, subPath, context);
+  }
+
+  /**
+   * Deletes a document with given ID in the given namespace and collection at the specified
+   * sub-path. Any previously existing sub-document at the given path will be removed.
+   *
+   * @param db {@link DocumentDB} to write in
+   * @param namespace Namespace
+   * @param collection Collection name
+   * @param documentId The ID of the document to delete
+   * @param subPath Sub-path of the document to delete. If empty will delete the whole doc.
+   * @param context Execution content
+   * @return Flag representing if the operation was success.
+   */
+  public Single<Boolean> deleteDocument(
+      DocumentDB db,
+      String namespace,
+      String collection,
+      String documentId,
+      List<String> subPath,
+      ExecutionContext context) {
+
+    return Single.defer(
+            () -> {
+              // authentication for writing before anything
+              authorizeWrite(db, namespace, collection, Scope.DELETE);
+
+              // pre-process to support array elements
+              List<String> subPathProcessed = processSubDocumentPath(subPath);
+
+              // call write document
+              return writeService.deleteDocument(
+                  db.getQueryExecutor().getDataStore(),
+                  namespace,
+                  collection,
+                  documentId,
+                  subPathProcessed,
+                  context);
+            })
+        // TODO should we extract applied here and return that
```

Comment:
Does applied actually return whether data was modified? I had assumed that running a delete that matches no rows would still return `applied` as `true`

If that's not the case, then we could actually differentiate between "the deletion occurred on existing doc" (204 no content) and "the deletion didn't occur because there was nothing there" (404 not found)

Reply:
I am actually not sure, however seems that this can be controlled with the `IF EXISTS`: https://docs.datastax.com/en/cql-oss/3.3/cql/cql_reference/cqlDelete.html? What do you say, add this to the deletion and correctly go for the different HTTP code? Seems cool? Not sure if this will be considered as API breaking?

Reply:
Not sure if it matters but for idempotency perspective returning success in both cases is usually considered better I think: success means that Table no longer exists and so calling DELETE (for timeouts etc) multiple times is safe.
404 can be overloaded as well since it can signal bad URL amongst other things.

But this is a good question as it applies to other APIs as well: ideally we should follow the same guideline.

Reply:
OK, I will leave the TODO for now and keep this things as is.. We can change in the V2 if we want..